ethcore/src/block.rs

```diff
@@ -116,8 +116,6 @@ pub struct ExecutedBlock {
 	pub traces: Tracing,
 	/// Hashes of last 256 blocks.
 	pub last_hashes: Arc<LastHashes>,
-	/// Finalization flag.
-	pub is_finalized: bool,
```

Comment:
this flag is completely redundant and is never set to true

Reply:
I think we're going to use it #9113? cc @andresilva 

I think we won't implement Casper again in the short terms, so some of the #8401 refactorings might use a better interface. But I do think allowing finalized block is a feature that we may use in other engines.

Reply:
Ah right. Aura only finalize ancestry blocks so it only needs `BlockChain::mark_finalized`. However, do we want to support instant finalization for some of our engines in the future? That's where we will need this flag.

I'm not sure whether POC-1 engine (which will replace the current Tendermint engine) will need this. [The spec](https://github.com/w3f/polkadot-spec/blob/master/spec.md#consensus) do mention instant finality. But I'm not really familiar with the algorithm.

Reply:
> However, do we want to support instant finalization for some of our engines in the future? That's where we will need this flag.

Than we should add this flag in the future. Imo, we shouldn't keep in our codebase anything that is not used right now

Reply:
I can't say I fully agree with this, but it sounds okay to me. :+1: 

Based on the above rationale, however, another thing we need to remove is metadata (https://github.com/paritytech/parity/blob/21d3de236005732253dee442cfbcb52ebcbbc1e7/ethcore/src/block.rs#L119 and https://github.com/paritytech/parity/blob/21d3de236005732253dee442cfbcb52ebcbbc1e7/ethcore/src/blockchain/extras.rs#L155). Also introduced by #8401, but not used by current engines.

Reply:
metadata removed! :)

Reply:
We can remove this field as well:

https://github.com/paritytech/parity-ethereum/blob/a598be7b6e3b0d35006d6c25d99e966b002749c7/ethcore/src/blockchain/extras.rs#L156

Metadata or finalization has not been set by anyone, so we can just change the `!is_short_version` to be 5 RLP fields. :)

Reply:
done ;)